scapy/layers/netflow.py

```diff
@@ -341,45 +383,87 @@ def dispatch_hook(cls, _pkt=None, *args, **kargs):
         if _pkt:
             if _pkt[:2] == b"\x00\x01":
                 return NetflowOptionsFlowsetV9
+            if _pkt[:2] == b"\x00\x00":
+                return NetflowFlowsetV9
         return cls
-    
-    def post_dissection(self, pkt):
-        # We need the whole packet to be dissected to access field def in NetflowFlowsetV9
+
+def netflowv9_defragment(plist):
+    """Process all NetflowV9 Packets to match IDs of the DataFlowsets with the Headers.
+    plist: the list of mixed NetflowV9 packets."""
+    # We need the whole packet to be dissected to access field def in NetflowFlowsetV9 or NetflowOptionsFlowsetV9
+    packet_list = plist.filter(lambda x: x.haslayer(NetflowFlowsetV9) or x.haslayer(NetflowOptionsFlowsetV9))
```

Comment:
Don't use `.filter()` + `lambda`, prefer something like `packet_list = [pkt for pkt in plist if [...]]`

Reply:
Updated